Dashboard/app/js/lib/components/devices/AssignmentsEditView/index.jsx

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FolderSurveySelectorView from 'akvo-flow/components/selectors/FolderSurveySelector';
+import FormSelectorView from 'akvo-flow/components/selectors/FormSelector';
+import DeviceGroupSelectorView from 'akvo-flow/components/selectors/DeviceSelector';
+import AssignmentDetails from './AssignmentDetails';
+
+export default class AssignmentsEditView extends React.Component {
+  state = {
+    assignmentName: this.props.inputValues.assignmentName,
+    startDate: this.props.inputValues.startDate,
+    expireDate: this.props.inputValues.toDate,
+    nameValidationMsg: '',
+    expireDateMsg: '',
+    startDateMsg: '',
+  }
+
+  // lifecycle methods
+  componentDidMount() {
+    this.validateAssignmentName(this.props.inputValues.assignmentName);
+    this.validateDate('startDateMsg', this.props.inputValues.startDate || '');
+    this.validateDate('expireDateMsg', this.props.inputValues.toDate || '');
+  }
+
+  // event handlers
+  onChangeState = (key, value) => {
+    if (key === 'assignmentName') this.validateAssignmentName(value);
+    if (key === 'startDate' || key === 'expireDate') this.validateDate(`${key}Msg`, value);
+    this.setState({ [key]: value });
+  }
+
+  onSubmit = () => {
+    const { assignmentName, startDate, expireDate } = this.state;
+
+    this.props.actions.onSubmit({ assignmentName, startDate, expireDate });
+  }
```

Comment:
Perhaps we can also add the validation just before submitting?

Reply:
Well, the problem here is that we don't want to add the FLOW variable to the react side

Reply:
I mean the same validation that you are performing in the `onChangeState` function. Does that require the FLOW variable?

Reply:
Oh, so rather than it being in the `onChange` method. It should just validate when the save button is clicked. I get it now?